### ISSUE TITLE
Can't pipe to head

### ISSUE THREAD

**Issue by sstadick** [theryangeary/choose#36]
Trying to pipe into head results in repeated `Failed to write to output: Broken pipe (os error 32)` messages.

**Comment by theryangeary** [comment]
Hi @sstadick, thanks for reporting this. This issue appears to be very similar to #34. Would you be able to checkout and build the latest code on the develop branch, and see if the issue persists? I am unable to reproduce it. If it does persist, please provide more info about your environment and the exact command you are running to produce the error. Thanks!

**Comment by sstadick** [comment]
I'm betting you are testing with a file smaller than your buffer size:

```
./target/release/choose -f ',' 1 8 19 -i ~/Downloads/all_train.csv | head
<... first ten lines ...> 
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
Failed to write to output: Broken pipe (os error 32)
... 
```

The `all_train.csv` is this: https://archive.ics.uci.edu/ml/machine-learning-databases/00347/all_train.csv.gz decompressed.

I pulled the `develop` branch and built it and it sill gives the same error. 

See https://github.com/theryangeary/choose/pull/37 for a fix.

**Comment by theryangeary** [comment]
Hi @sstadick, thanks for following through with testing, more details, and the code update. Sorry to have given you bad information, but I've realized that I actually never merged #35 into develop before. I've merged it now.

With the additional information you gave and a sufficiently large input file, I am able to reproduce the issue with the old code, and can confirm that with the newly updated develop branch, the issue is in fact resolved. Go ahead and pull the new changes on develop, and please report back in the event that you still see any issues. Thanks!
